### ISSUE TITLE
Can't pass `datatype` argument to `crop` inside `cropInputs`

### ISSUE THREAD

**Issue by CeresBarros** [PredictiveEcology/reproducible#195]
I'm trying to pass `datatype` to `raster::crop` via `cropInputs`, but the argument is turned into `NULL` here:
```
dots <- list(...)
Browse[2]> dots
$datatype
[1] "INT1U"

dots[.formalsNotInCurrentDots("crop", ...)] <- NULL

Browse[2]> .formalsNotInCurrentDots("crop", ...)
[1] "datatype"
```    

This then leads to the error (when the raster cannot be processed in memory:
```
Browse[2]> do.call(raster::crop, args = append(list(x = x, 
+                 y = cropExtentRounded, filename = paste0(tempfile(tmpdir = tmpDir()), 
+                   ".tif")), dots))
Error in if (!datatype %in% c("LOG1S", "INT1S", "INT2S", "INT4S", "INT1U",  : 
  argument is of length zero
  ```

**Comment by eliotmcintire** [comment]
Classic problem of getting the formals of an S4 method... I found this 
https://r.789695.n4.nabble.com/Q-Get-formal-arguments-of-my-implemented-S4-method-td4702420.html 

```
methodFormals <- function(f, signature = character()) {
    fdef <- getGeneric(f)
    method <- selectMethod(fdef, signature)
    genFormals <- base::formals(fdef)
    b <- body(method)
    if(is(b, "{") && is(b[[2]], "<-") && identical(b[[2]][[2]], as.name(".local"))) {
        local <- eval(b[[2]][[3]])
        if(is.function(local))
            return(formals(local))
        warning("Expected a .local assignment to be a function. Corrupted method?")
    }
    genFormals
}
```
which could be used inside that `.formalsNotInDots` function.

**Comment by eliotmcintire** [comment]
Should be fixed with #197

**Comment by CeresBarros** [comment]
Tested and does seem fixed :)
